blaze/expr/core.py

```diff
@@ -404,6 +404,7 @@ def subterms(x):
     yield x
 
 
+@memoize_mutable
 def subs(o, d):
```

Comment:
Sorry, I saw the OP and the other pr, seems like this is actually a big win. I still wonder if we can speed it up further by avoiding pickle.